pkg/instancetype/vm/controller/controller.go

```diff
@@ -0,0 +1,226 @@
+	expandHandler
+	upgradeHandler
+
+	// FIXME(lyarwood): Hackaround as all find implementations provide a function named Find
+	Finder           instancetypeFindHandler
+	PreferenceFinder preferenceFindHandler
+
+	clientset     kubecli.KubevirtClient
+	clusterConfig *virtconfig.ClusterConfig
+	recorder      record.EventRecorder
+}
+
+func New(
+	instancetypeStore, clusterInstancetypeStore, preferenceStore, clusterPreferenceStore, revisionStore cache.Store,
+	virtClient kubecli.KubevirtClient, clusterConfig *virtconfig.ClusterConfig, recorder record.EventRecorder,
+) *controller {
+	finder := find.NewSpecFinder(instancetypeStore, clusterInstancetypeStore, revisionStore, virtClient)
+	prefFinder := preferencefind.NewSpecFinder(preferenceStore, clusterPreferenceStore, revisionStore, virtClient)
+	return &controller{
+		Finder:           finder,
+		PreferenceFinder: prefFinder,
+		applyVMIHandler:  apply.NewVMIApplier(),
+		applyVMHandler:   apply.NewVMApplier(finder, prefFinder),
+		storeHandler:     revision.New(instancetypeStore, clusterInstancetypeStore, preferenceStore, clusterPreferenceStore, virtClient),
+		expandHandler:    expand.New(clusterConfig, finder, prefFinder),
+		upgradeHandler:   upgrade.New(revisionStore, virtClient),
+		clientset:        virtClient,
+		clusterConfig:    clusterConfig,
+		recorder:         recorder,
+	}
+}
+
+func (c *controller) Find(vm *virtv1.VirtualMachine) (*v1beta1.VirtualMachineInstancetypeSpec, error) {
+	return c.Finder.Find(vm)
+}
+
+func (c *controller) FindPreference(vm *virtv1.VirtualMachine) (*v1beta1.VirtualMachinePreferenceSpec, error) {
+	return c.PreferenceFinder.Find(vm)
+}
+
+const (
+	storeControllerRevisionErrFmt   = "Error encountered while storing Instancetype ControllerRevisions: %v"
+	upgradeControllerRevisionErrFmt = "error encountered while upgrading instancetype.kubevirt.io controllerRevisions: %v"
+	cleanControllerRevisionErrFmt   = "error encountered cleaning controllerRevision %s after successfully expanding VirtualMachine %s: %v"
+)
+
+func (c *controller) Sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) (*virtv1.VirtualMachine, error) {
+	if vm.Spec.Instancetype == nil && vm.Spec.Preference == nil {
+		return vm, nil
+	}
+
+	referencePolicy := c.clusterConfig.GetInstancetypeReferencePolicy()
+	switch referencePolicy {
+	case virtv1.Reference:
+		// Ensure we have controllerRevisions of any instancetype or preferences referenced by the VM
+		if err := c.Store(vm); err != nil {
+			log.Log.Object(vm).Errorf(storeControllerRevisionErrFmt, err)
+			c.recorder.Eventf(vm, k8score.EventTypeWarning, common.FailedCreateVirtualMachineReason, storeControllerRevisionErrFmt, err)
+			// FIXME(lyarwood) Align storeControllerRevisionErrFmt with the other errors and use lower case
+			//nolint:stylecheck
+			return vm, common.NewSyncError(fmt.Errorf(storeControllerRevisionErrFmt, err), common.FailedCreateVirtualMachineReason)
+		}
+	case virtv1.Expand, virtv1.ExpandAll:
```

Comment:
Is a default case required?

Reply:
No `GetInstancetypeReferencePolicy` always returns `virtv1.Reference` as the default.